pkg/cache/sql/attachdriver/driver.go

```diff
@@ -0,0 +1,62 @@
+/*
+Package attachdriver provides a sqlite driver that wraps any other driver and attaches a database at a given path when a connection is opened.
+*/
+package attachdriver
+
+import (
+	"database/sql"
+	"database/sql/driver"
+	"fmt"
+
+	"modernc.org/sqlite"
+)
+
+// AttachDriver uses an underlying driver to open the connection and attaches a database
+// to that connection using path.
+type AttachDriver struct {
+	path string
+	d    driver.Driver
+}
+
+// Register registers the attachdriver with the given path with go's sql package
+func Register(path string) {
+	d := &AttachDriver{
+		path: path,
+		d:    &sqlite.Driver{},
+	}
+	sql.Register("attach_sqlite", d)
+}
+
+// Open opens a connection from the underlying driver and
+// attaches a database as db2 using the underlying path.
+func (ad AttachDriver) Open(name string) (driver.Conn, error) {
+	c, err := ad.d.Open(name)
+	if err != nil {
+		return nil, err
+	}
+
+	q := fmt.Sprintf(`ATTACH DATABASE '%s' AS db2;`, ad.path)
```

Comment:
This is prone to SQL injection. If path is `file:/tmp/sqlite' as DB2;ATTACH DATABASE "/tmp/target" AS tg; CREATE TABLE tg.abc (data text); INSERT INTO tg.abc (data) VALUES ("123456");--`, an arbitrary sqlite db (`/tmp/target`) will be created with whatever values defined within the `path`.

Let's add some validation and make this value quite strict.

Reply:
Fixed